src/inject/dynamic-theme/index.ts

```diff
@@ -299,6 +313,13 @@ function watchForUpdates() {
         }
         newManagers.forEach((manager) => manager.watch());
         stylesToRestore.forEach((style) => styleManagers.get(style).restore());
+    }, (shadowroot) => {
+        const shadowRootVariable = new Map<string, string>();
+        forEach(shadowroot.adoptedStyleSheets, (stylesheet) => {
+            getCSSVariables(stylesheet.cssRules).forEach((value, key) => shadowRootVariable.set(key, value));
+        });
+        updateVariables(shadowRootVariable);
```

Comment:
Shadow CSS variables should not affect globals. Do something that will join global and shadow variables in `render` method.